components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/GeoLocationBasedServiceImpl.java

```diff
@@ -300,6 +381,47 @@ public Response getGeoAlerts(@PathParam("deviceId") String deviceId,
         }
     }
 
+    @Path("alerts/{alertType}")
+    @GET
+    @Consumes("application/json")
+    @Produces("application/json")
+    public Response getGeoAlertsForGeoClusters(@PathParam("alertType") String alertType) {
+        try {
+
+            // this is the user who initiates the request
+            String authorizedUser = MultitenantUtils.getTenantAwareUsername(
+                    CarbonContext.getThreadLocalCarbonContext().getUsername()
+            );
+
+            GeoLocationProviderService geoService = DeviceMgtAPIUtils.getGeoService();
+
+            if (GeoServices.ALERT_TYPE_WITHIN.equals(alertType)) {
+                List<GeoFence> alerts = geoService.getWithinAlerts();
+                return Response.ok().entity(alerts).build();
+            } else if (GeoServices.ALERT_TYPE_EXIT.equals(alertType)) {
+                List<GeoFence> alerts = geoService.getExitAlerts();
+                return Response.ok().entity(alerts).build();
+            } else if (GeoServices.ALERT_TYPE_SPEED.equals(alertType)) {
+                String result = geoService.getSpeedAlerts();
+                return Response.ok().entity(result).build();
+            } else if (GeoServices.ALERT_TYPE_PROXIMITY.equals(alertType)) {
+                String result = geoService.getProximityAlerts();
+                return Response.ok().entity(result).build();
+            } else if (GeoServices.ALERT_TYPE_STATIONARY.equals(alertType)) {
+                List<GeoFence> alerts = geoService.getStationaryAlerts();
+                return Response.ok().entity(alerts).build();
+            } else if (GeoServices.ALERT_TYPE_TRAFFIC.equals(alertType)) {
+                List<GeoFence> alerts = geoService.getTrafficAlerts();
+                return Response.ok().entity(alerts).build();
+            }
+            return null;
```

Comment:
Shall we throw an error when none of the above matches?

Reply:
@rasika and @Arcane94 Can't we replace the above block with a switch?

Reply:
@rasika  and @Kamidu  Fixed c653051